### PR TITLE
PPR UI new registration add save draft.

### PR DESCRIPTION
**Pull request by doug-lovett**
*Issue #:* /bcgov/entity#7262

*Description of changes:*
Add new financing statement save draft functionality. UI submit on button click pending as a separate PR.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/interfaces/ppr-api-interfaces/registration-interfaces.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  type: string, // One of enum APIRegistrationTypes.
+  clientReferenceId?: string, // AKA folio max length 20.
+  documentId?: string, // Optional draft ID if draft created.
+  registrationDescription?: string, // Returned on creation.
+  registrationAct?: string, // Returned on creation.
+  registeringParty: PartyIF,
+  securedParties: PartyIF[],
+  debtors: PartyIF[],
+  vehicleCollateral?: VehicleCollateralIF[], // Either vehicle or general collatera is required.
+  generalCollateral?: GeneralCollateralIF[], // Max length 4000, currently only 1 for a new registration.
```
Comment (jordiwes) [inline ppr-ui/src/interfaces/ppr-api-interfaces/registration-interfaces.ts:21]: wondering if this should be an array...

File: ppr-ui/src/interfaces/ppr-api-interfaces/registration-interfaces.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  trustIndenture?: boolean, // Conditionally required by registration type.
+  lifeInfinite?: boolean,
+  lifeYears?: number, // 1..25 if not lifeInfinite, otherwise 0.
+  lienAmount?: string, // RL only
+  surrenderDate?: string, // RL only
```
Comment (jordiwes) [inline ppr-ui/src/interfaces/ppr-api-interfaces/registration-interfaces.ts:26]: should dates be a date type?

File: ppr-ui/src/interfaces/ppr-api-interfaces/vehicle-collateral-interface.ts [hunk 3]
```diff
@@ -6,5 +6,6 @@ export interface VehicleCollateralIF {
   year?: number, // Optional
   make?: string, // Optional
   model?: string, // Optional
-  manufacturedHomeRegistrationNumber?: string // Conditional: only returned on MHR number search.
+  manufacturedHomeRegistrationNumber?: string, // Conditional: only returned on MHR number search.
+  vehicleId?: number // System generated used for amendment/change registrations.
```
Comment (jordiwes) [inline ppr-ui/src/interfaces/ppr-api-interfaces/vehicle-collateral-interface.ts:10]: I have a property called just "id" in my code.... I can change it to vehicleId

File: ppr-ui/src/utils/registration-helper.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+/** Save or update the current financing statement. Data to be saved is in the store state model. */
+export async function saveFinancingStatementDraft (stateModel:StateModelIF): Promise<DraftIF> {
+  const registrationType: RegistrationTypeIF = stateModel.registrationType
+  // console.log('registrationType: ' + JSON.stringify(registrationType))
+  var error:ErrorIF = null
+  var draft:DraftIF = stateModel.draft
+  // console.log('draft: ' + JSON.stringify(draft))
+  draft.type = DraftTypes.FINANCING_STATEMENT
+  var statement:FinancingStatementIF = draft.financingStatement
+  console.log('statement: ' + JSON.stringify(statement))
```
Comment (jordiwes) [inline ppr-ui/src/utils/registration-helper.ts]: keep the console.log?

File: ppr-ui/tests/unit/test-data/mock-new-registration.ts [hunk 5]
```diff
@@ -13,3 +26,145 @@ export const mockedError: ErrorIF = {
...
+  }
+]
+
+export const mockedVehicleCollateral1: Array<VehicleCollateralIF> = [
+  {
+    type: VehicleTypes.MOTOR_VEHICLE,
+    serialNumber: 'KM8J3CA46JU622994',
+    year: 2018,
+    make: 'HYUNDAI',
+    model: 'TUSCON'
```
Comment (jordiwes) [inline ppr-ui/tests/unit/test-data/mock-new-registration.ts:51]: probably need vehicleId here

File: ppr-ui/src/interfaces/ppr-api-interfaces/registration-interfaces.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+export interface PaymentIF {
+  invoiceId: string, // Unique identifier of the payment transaction.
+  receipt: string // Payment API URI to obtain the payment receipt.
+}
+
+// Financing Statement interface. All dates/date time properties are in the ISO 8601 format YYYY-MM-DDThh:mm:ssTZD.
+export interface FinancingStatementIF {
+  type: string, // One of enum APIRegistrationTypes.
+  clientReferenceId?: string, // AKA folio max length 20.
+  documentId?: string, // Optional draft ID if draft created.
```
Comment (jordiwes) [inline ppr-ui/src/interfaces/ppr-api-interfaces/registration-interfaces.ts:14]: I don't see anywhere where this documentId is set...

File: ppr-ui/src/utils/registration-helper.ts (hunk 4 above)
Comment (doug-lovett) [reply ppr-ui/src/utils/registration-helper.ts]: Thanks, I have removed it.

File: ppr-ui/src/interfaces/ppr-api-interfaces/registration-interfaces.ts (hunk 6 above)
Comment (doug-lovett) [reply ppr-ui/src/interfaces/ppr-api-interfaces/registration-interfaces.ts:14]: I think all your type questions are answered by the looking at the API spec:
https://bcregistries-daxiom.web.app/ppr-api
ID's are generated by the API and returned in responses.

**Pull request merged**